File: docs/test/coding_tutorial/task_answers/Task 4.0 (ifelse)/ifelse.cpp
// Your task is to modify the ifelse C++ example program so that it gets user input to change on to either true or false similar to the ifelse Java example. 
#include <iostream>
int main(){
    bool on{};
    std::string light_switch_state{};
    std::cout << "Enter on to turn on the switch and off to turn off the switch" << "\n";
    std::cin >> light_switch_state;
    if(light_switch_state=="on"){
        on = true;
        std::cout << "Light bulb turned on" << "\n";
    }
    else{
        on =false;
        std::cout << "Light bulb turned off" << "\n";
    }
    return 0;
}

File: docs/test/coding_tutorial/java/ifelse/ifelse.java
import java.util.Scanner;
public class ifelse {  // In Java, when you have a public class, the filename must match the class name. Filename: ifelse.java
    public static void main(String[] args){ // this is the main function
        boolean light_bulb;

        // Create a Scanner object
        Scanner scanner = new Scanner(System.in);
        
        // Prompt the user
        System.out.println("Enter on to turn on the switch and off to turn off the switch");

        // get user input
        String light_switch_state = scanner.nextLine(); // gets the typed text from the next line

        // anything other than "on" will turn off the light
        if(light_switch_state.equals("on")){ // test if light_switch_state equals "on"
            light_bulb = true; // set light_bulb to on
            System.out.println("Light bulb turned on"); // output text to the terminal
        }
        else{
            light_bulb = false; // turn off light_bulb
            System.out.println("Light bulb turned off"); // output text to the terminal
        }

        scanner.close(); // delete scanner object
    }
}